pkg/controller/seed-controller-manager/cni-application-installation-controller/controller.go

```diff
@@ -168,13 +170,24 @@ func (r *Reconciler) Reconcile(ctx context.Context, request reconcile.Request) (
 	}
 	if result == nil {
 		result = &reconcile.Result{}
+
+		if r.systemAppEnforceInterval != 0 {
+			// Reconciliation was successful, but requeue in systemAppEnforceInterval minutes if set.
+			// We do this to make sure that ApplicationInstallation in the user cluster is not modified in a wrong way / deleted accidentally.
+			// Even though it is protected by a webhook, not all unwanted modifications can be easily prevented there.
+			result.RequeueAfter = time.Duration(r.systemAppEnforceInterval) * time.Minute
+		}
```

Comment:
This will only be triggered if `result == nil` if I see correctly, which seems to happen only when `r.reconcile(ctx, log, cluster)` fails to remove annotations. Is that the intention behind it? Or am I missing something?

Reply:
`r.reconcile` was returning `nil` result on success, and that was when we wanted to reconcile.
But I moved this logic directly at the end of `r.reconcile` so that it is better readable.

Reply:
Ah, see, I missed that last return for whatever reason. Sorry!